server/preprocessing/other-scripts/vis_layout.R

```diff
@@ -18,130 +18,189 @@ vis_layout <- function(text, metadata, max_clusters=15, maxit=500, mindim=2, max
   print("calc matrix")
   result <- create_tdm_matrix(metadata, text);
   metadata_full_subjects = result$metadata_full_subjects
-  
+  metadata_full_subjects_tfidf_top5 <- create_node_names(metadata_full_subjects)
+
   print("normalize matrix")
   normalized_matrix <- normalize_matrix(result$tdm_matrix);
-  
+
   print("create clusters")
   clusters <- create_clusters(normalized_matrix, max_clusters=15);
   layout <- create_ordination(normalized_matrix, maxit=500, mindim=2, maxdim=2)
   output <- create_output(clusters, layout, metadata_full_subjects)
-  
+
   return(output)
-  
+
 }
 
 create_tdm_matrix <- function(metadata, text, sparsity=1, lang="english") {
   m <- list(content = "content", id = "id")
-  
+
   myReader <- readTabular(mapping = m)
-  
+
   (corpus <- Corpus(DataframeSource(text), readerControl = list(reader = myReader)))
-  
+
   corpus <- tm_map(corpus, removePunctuation)
-  
-  corpus <- tm_map(corpus, content_transformer(tolower))
-  
+
   corpus <- tm_map(corpus, removeWords, stopwords(lang))
-  
-  metadata_full_subjects <- replace_keywords_if_empty(corpus, metadata)
-  
+
   corpus <- tm_map(corpus, stripWhitespace)
-  
+
+  metadata_full_subjects <- replace_keywords_if_empty(corpus, metadata)
+
+  # corpus <- tm_map(corpus, content_transformer(tolower))
+
   corpus_unstemmed = corpus
-  
+
   corpus <- tm_map(corpus, stemDocument)
-  
+
   tdm <- TermDocumentMatrix(corpus)
-  
+
   if(sparsity < 1) {
     tdm <- removeSparseTerms(tdm, sparsity)
   }
-  
+
   tdm_matrix = t(as.matrix(tdm))
-  
+
   return(list(tdm_matrix = tdm_matrix, metadata_full_subjects = metadata_full_subjects))
 }
 
+replace_keywords_if_empty <- function(corpus, metadata) {
+
+  missing_subjects = which(lapply(metadata$subject, function(x) {nchar(x)}) <= 1)
 
+  candidates = mapply(paste, metadata$title[missing_subjects], metadata$paper_abstract[missing_subjects])
+  candidates = lapply(candidates, function(x)paste(removeWords(x, stopwords('english')), collapse=""))
```

Comment:
language is now a parameter, but within function-scope, not global. may need improvement